### PR TITLE
Change `contrib-tests.yml` tests to use `--skip-openai`

### LINKED ISSUE #1112: [Issue]: test in `contrib-tests.yml` should also use `--skip-openai` to skip the openai test
### Describe the issue

Test in `contrib-tests.yml` should also use `--skip-openai` instead of `pip uninstall -y openai` to skip the openai test.

This issue is related to #1088 and #1097

### Steps to reproduce

_No response_

### Screenshots and logs

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by maxim-saplin**
## Why are these changes needed?

`contrib-tests.yml`  workflow used `pip uninstall -y openai` trick to skip the tests depending on OpenAI. Changed it to use pytest argument `--skip-openai` instead

## Related issue number

Closes #1112

## Notes

As of now the tests in `contrib-tests.yml` are mostly skipped cause they depend on OpenAI (~70-80% of individual tests are skipped). It makes sense to tidy up the file and remove all those OpenAI dependent tests as they don't do anything. Here're the breakdown for individual files:

```
RetrieveChatTest

	• test/test_retrieve_utils.py
	• [x][Always Skipped] test/agentchat/contrib/test_retrievechat.py
	• test/agentchat/contrib/test_qdrant_retrievechat.py


[x]CompressionTest
	
	• [x][Always Skipped]test/agentchat/contrib/test_compressible_agent.py


[x]GPTAssistantAgent
	
	• [x][Always Skipped]test/agentchat/contrib/test_gpt_assistant.py


[x]TeachableAgent

	• [x][Always Skipped]test/agentchat/contrib/test_teachable_agent.py


[x]LMMTest
	• test/agentchat/contrib/test_img_utils.py
	• test/agentchat/contrib/test_lmm.py
[x]test/agentchat/contrib/test_llava.py - depends on LLAVA imports
```

## Checks

- [ N/A] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [N/A ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by afourney — APPROVED** [review]
LGTM, but may conflict with #1072 since several tests needed to be corrected in that PR as well.

File: test/agentchat/contrib/test_qdrant_retrievechat.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import pytest
 from autogen.agentchat.contrib.retrieve_assistant_agent import RetrieveAssistantAgent
 from autogen import config_list_from_json
+from conftest import skip_openai
```
Comment (sonichi) [inline test/agentchat/contrib/test_qdrant_retrievechat.py:6]: The import will fail if we directly run this python file.

File: test/agentchat/contrib/test_retrievechat.py [hunk 2]
```diff
@@ -2,6 +2,7 @@
 import os
 import sys
 import autogen
+from conftest import skip_openai
```
Comment (sonichi) [inline test/agentchat/contrib/test_retrievechat.py:5]: same

**Review by sonichi — COMMENTED** [review]
There's a small issue.

File: test/agentchat/contrib/test_qdrant_retrievechat.py (hunk 1 above)
Comment (rickyloynd-microsoft) [reply test/agentchat/contrib/test_qdrant_retrievechat.py:6]: Would the fix from #1144 work here as well?

```
sys.path.append(os.path.join(os.path.dirname(__file__), "../.."))
from conftest import skip_openai  # noqa: E402

```

**Comment by maxim-saplin** [comment]
@sonichi, I've modified both files, please check

**Pull request merged**